src/pages/index.js

```diff
@@ -20,100 +24,178 @@ import {
-const previewImageEl = previewModal.querySelector(".modal__preview-image");
+const editModalSubmitButton = editModalForm.querySelector(".modal__button");
 
 /*-- Profile Selectors --*/
+const profileAvatarForm = document.querySelector("#profile-modal-form");
+const profilePenIcon = document.querySelector(`#profile__avatar-pen-icon`);
 const profileEditButton = document.querySelector(".profile__button-edit");
 const profileAddButton = document.querySelector(".profile__button-add");
-const profileName = document.querySelector(".profile__name");
-const profileBio = document.querySelector(".profile__bio");
+const avatarModalSubmitButton =
+  profileAvatarForm.querySelector(".modal__button");
 
-const items = cardsData;
-const section = new Section({ items, renderer: createCard }, ".card__list");
-section.renderItems();
+/*-- Trash Icon Selectors --*/
+const trashModal = document.querySelector("#trash-modal");
+const trashModalSubmitButton = trashModal.querySelector(".modal__button");
 
 /*---------------------------------------------------*/
-/*                     Functions                     */
+/*                 Event Handlers                    */
 /*---------------------------------------------------*/
 
-function createCard(cardData) {
-  const name = cardData.name;
-  const altName = cardData.name;
-  const link = cardData.link;
-  const card = new Card(
-    { name, altName, link },
-    "#card-template",
-    handleImageClick
-  );
-  return card.getView();
+function handleAvatarFormSubmit(formValues) {
+  const profilePicObject = api.patchProfileAvatar(formValues.url);
+  changeSubmitButton(avatarModalSubmitButton);
+  profilePicObject.then((object) => {
+    avatarPopup.handleAvatarChange(object.avatar);
+  });
+  avatarPopup.close();
+  profileAvatarFormValidator.disableButton();
 }
 
-/*---------------------------------------------------*/
-/*                 Event Handlers                    */
-/*---------------------------------------------------*/
-
 function handleProfileFormSubmit(formValues) {
+  api.patchProfile(formValues.name, formValues.bio);
+  changeSubmitButton(editModalSubmitButton);
   userInfo.setUserInfo(formValues.name, formValues.bio);
   profilePopup.close();
   editProfileFormValidator.disableButton();
 }
 
 function handleAddCardFormSubmit(formValues) {
   const name = formValues.title;
-  const altName = formValues.title;
   const link = formValues.url;
-  const cardElement = createCard({ name, altName, link });
-  section.addItem(cardElement);
+  const cardApiObject = api.postCards({ name, link });
+  changeSubmitButton(addModalSubmitButton);
+  cardApiObject.then((object) => {
+    const element = createCard(object);
+    section.addItem(element);
+  });
   cardPopup.close();
 }
 
 function handleImageClick(data) {
   popupImg.open(data);
+  popupImg.setEventListeners();
+}
+
+function handleConfirmModal(data) {
+  trashModalSubmitButton.textContent = "Yes";
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.